code/modules/antagonists/eldritch_cult/eldritch_items.dm

```diff
@@ -180,3 +189,219 @@
+		if(suittoggled)
+			cloak.max_items = 3
+			to_chat(carbon_user,"<span class='notice'>The light shifts around you making the cloak invisible!</span>")
+		else
+			cloak.max_items = 4
+	else
+		to_chat(carbon_user,"<span class='danger'>You can't force the hood onto your head!</span>")
+
+	item_flags = suittoggled ? EXAMINE_SKIP : ~EXAMINE_SKIP
+
+/obj/item/clothing/mask/void_mask
+	name = "Abyssal Mask"
+	desc = "Mask created from the suffering of existance, you can look down it's eyes, and notice something gazing back at you."
+	icon_state = "mad_mask"
+	inhand_icon_state = "mad_mask"
+	w_class = WEIGHT_CLASS_SMALL
+	flags_cover = MASKCOVERSEYES
+	resistance_flags = FLAMMABLE
+	flags_inv = HIDEFACE|HIDEFACIALHAIR
+	var/mob/living/carbon/human/local_user
+
+/obj/item/clothing/mask/void_mask/equipped(mob/user, slot)
+	. = ..()
+
+	if(ishuman(user) && user.mind && slot == ITEM_SLOT_MASK)
+		local_user = user
+		START_PROCESSING(SSobj,src)
+		if(IS_HERETIC(user) || IS_HERETIC_MONSTER(user))
+			return
+		ADD_TRAIT(src, TRAIT_NODROP, CLOTHING_TRAIT)
+
+/obj/item/clothing/mask/void_mask/dropped(mob/M)
+	. = ..()
+	local_user = null
+	STOP_PROCESSING(SSobj,src)
+	REMOVE_TRAIT(src, TRAIT_NODROP, CLOTHING_TRAIT)
+
+/obj/item/clothing/mask/void_mask/process(delta_time)
+	if(!local_user)
+		return PROCESS_KILL
+
+	if((IS_HERETIC(local_user) || IS_HERETIC_MONSTER(local_user)) && HAS_TRAIT(src,TRAIT_NODROP))
+		REMOVE_TRAIT(src, TRAIT_NODROP, CLOTHING_TRAIT)
+
+	for(var/mob/living/carbon/human/human_in_range in spiral_range(9,local_user))
+		if(IS_HERETIC(human_in_range) || IS_HERETIC_MONSTER(human_in_range))
+			continue
+
+		SEND_SIGNAL(human_in_range,COMSIG_DRAIN_SANITY,rand(-1,-10))
+
+		if(DT_PROB(10,delta_time))
+			human_in_range.emote(pick("giggle","laugh"))
+			human_in_range.adjustStaminaLoss(10)
+
+		if(DT_PROB(25,delta_time))
+			human_in_range.Dizzy(5)
+
+		if(DT_PROB(40,delta_time))
+			human_in_range.Jitter(5)
+
+		if(DT_PROB(60,delta_time))
+			human_in_range.hallucination += 5
+
+/obj/item/melee/rune_knife
+	name = "Carving Knife"
+	desc = "Cold Steel, pure, perfect, this knife can carve the floor in many ways, but only few can evoke the dangers that lurk beneath reality."
+	icon = 'icons/obj/eldritch.dmi'
+	icon_state = "rune_carver"
+	flags_1 = CONDUCT_1
+	sharpness = SHARP_EDGED
+	w_class = WEIGHT_CLASS_SMALL
+	force = 10
+	throwforce = 5
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb_continuous = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "rends")
+	attack_verb_simple = list("attack", "slash", "stab", "slice", "tear", "lacerate", "rip", "dice", "rend")
+	var/static/list/blacklisted_turfs = typecacheof(list(/turf/closed,/turf/open/space,/turf/open/lava))
+	var/drawing = FALSE
+	var/list/current_runes = list()
+	var/max_rune_amt = 3
+	var/datum/action/innate/rune_shatter/linked_action
```

Comment:
Document these variables and properly delete and null `linked_action` on destruction
Also handle `current_runes` references

Reply:
the destruction of runes is already handled in carve_rune. Basically it just runs over the loop clearing any runes that are QDELETED or just nulls and subtracts them from the list.